### PR TITLE
ci: Added a PR template for contributions to the repository - PLT-9507

### PR DESCRIPTION
**Pull request by RSoulatIOHK**
Added a template for Pull Request with Checklist to ensure a smoother process and outside contributions.

Ref: PLT-9507

## Summary by CodeRabbit

- **Chores**
	- Introduced a standardized template for creating pull requests to improve consistency and quality of pull request descriptions.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Pull request template
+
+## Description
+
+Please include a summary of the change. You can rely on CodeRabbitAI to fill in the details of the changes.
```
[the remark this answers is not in the document]
Comment (RSoulatIOHK) [reply .github/pull_request_template.md:5]: That's quite the recommendation from CodeRabbitAI to remove the one line where we recommend it.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] New tests are added for the changes (if not, mention it in the PR decription). These may include:
+  - [ ] Unit tests,
+  - [ ] Integration tests,
+  - [ ] Property Based testing,
+  - [ ] End-to-end tests.
+- [ ] The new version builds and passes all tests. If not, please mention it in the PR description.
+- [ ] Self review of the code has been done.
+- [ ] Reviewer has been requested.
+- [ ] Ticket in Jira is in Review and sub-task for the reviewer has been created and assigned.
+<!-- -[ ] The contribution follows our CONTRIBUTING.md guidelines -->
```
[the remark this answers is not in the document]
Comment (RSoulatIOHK) [reply .github/pull_request_template.md:26]: We don't have a CONTRIBUTING.md file yet, I just want us to be aware that: we need one, and we want to add it to the tempalte when we do.

**Review by amnambiar — APPROVED** [review]
Approved!

**Comment by KJES4** [comment]
1. What would be the difference between a link to the issue and the link to the Jira ticket? We may also want to pause accepting this PR till after the group discussion about whether or not we keep using Jira moving forward. The content in this template will have to change depending on the outcome of that call.
- Include a link to the issue that is fixed.
- Include a reference to the Jira ticket(s) that have been worked on.

2. If we plan to keep using Jira and smart commits then the commit messages cannot be more than once sentence long.
- [ ] Commit messages broadly make sense and follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) format.

3. I think when we go public we can have both branches but for now we are just using one.
- [ ] PR targets master/develop

Other than those three points I think everything looks great!

**Pull request merged**